### DISCUSSION TITLE
Adonis suitable for building Rest API?

### DISCUSSION THREAD

**Discussion by vnugent** [adonisjs/core#1700] [Help] [answered]
After building a couple Rest API servers with plain Node I'm looking to adopt Adonis for my next Rest project.    Is Adonis primary used for building SSR or both, REST and SSR?

**Comment by thetutlage** [comment, marked as answer]
The question is a bit broader and hard to answer. It will be nice, if you can share the specific on what exactly you would want from an API server?

As far as SSR is concerned. It is just a matter of not installing the view package and returning JSON from your controller methods :)

**Reply by vnugent** [reply]
Traditional Rest API endpoints with GET, PUT, params parsing connecting to a relational DB model.

**Reply by thetutlage** [reply]
Everything you mentioned is there in AdonisJS. I recommend going to through the docs once :) https://preview.adonisjs.com/guides/quick-start

**Reply by vnugent** [reply]
I see now.  It wasn't clear in the landing page.
